### PR TITLE
Remove redundant code from FTPWindow.cpp (Fixes #55)

### LINKED ISSUE #55: Redundant code
There are some redundant sections in the codebase.

### PR DESCRIPTION
**Pull request by Zotyamester**
Modified version of #49

### CODE REVIEW

File: src/FTPWindow.cpp [hunk 1]
```diff
@@ -851,17 +849,14 @@ void TFTPWindow::CopyUrl()
...
-		fCurrentRemoteDir[fCurrentRemoteDir.Length() - 1] == '/') {
-
-		url << "ftp://" << fHost << ":" << fPort
-			<< fCurrentRemoteDir << intName;
-	} else {
-		url << "ftp://" << fHost << ":" << fPort
-			<< fCurrentRemoteDir << "/" << intName;
+	url << "ftp://" << fHost << ":" << fPort << fCurrentRemoteDir;
+	if (fCurrentRemoteDir.Length() <= 0
+		|| fCurrentRemoteDir[fCurrentRemoteDir.Length() - 1] != '/') {
```
Comment (owenca) [inline src/FTPWindow.cpp]: Remove the braces per Haiku Coding Guidelines.

File: src/FTPWindow.cpp [hunk 2]
```diff
@@ -851,17 +849,14 @@ void TFTPWindow::CopyUrl()
...
+		url << "/";
 	}
+	url << intName;
 
-	BMessage *clip = (BMessage *)NULL;
+	BMessage *clip = NULL;
```
Comment (owenca) [inline src/FTPWindow.cpp]:
```suggestion
	BMessage* clip = NULL;
```

Comment (Zotyamester) [reply src/FTPWindow.cpp]: Yeah, the problem is that the codebase does not follow the guidelines most of the time.
And so every pointer in the code is like 'Type *variablename' and not 'Type* variablename'
even though it should be.

Comment (Zotyamester) [reply src/FTPWindow.cpp]: So If I modified it, the coding style would not be uniform.

Comment (owenca) [reply src/FTPWindow.cpp]: It's already not consistent, even just in this file. I think the rule of thumb is that you fix the coding style (at least) for the lines you touch.

File: src/FTPWindow.cpp [hunk 3]
```diff
@@ -160,8 +160,9 @@ TFTPWindow::TFTPWindow(BRect frame, const char *name)
 	fRemoteFileView->AddStatusView(fItemCountView);
 
 	// CheckBox (Use This Connection)
-	const char* str = B_TRANSLATE("Use this connection");
-	fUseThisConnection = new BCheckBox("UseThisConnection", str, NULL, 0);
+	fUseThisConnection = new BCheckBox("UseThisConnection",
+									   B_TRANSLATE("Use this connection"),
```
Comment (Zotyamester) [inline src/FTPWindow.cpp]: In my editor it looks good, but when I view the changes on GitHub the indentation looks awkward.

Comment (owenca) [inline src/FTPWindow.cpp]:
```suggestion
		B_TRANSLATE("Use this connection"), NULL, 0);
```

File: src/FTPWindow.cpp [hunk 4]
```diff
@@ -160,8 +160,9 @@ TFTPWindow::TFTPWindow(BRect frame, const char *name)
...
+									   NULL, 0);
```
Comment (owenca) [inline src/FTPWindow.cpp]:
```suggestion
```

File: src/FTPWindow.cpp [hunk 5]
```diff
@@ -844,27 +843,20 @@ void TFTPWindow::Mkdir()
...
+		|| fCurrentRemoteDir[fCurrentRemoteDir.Length() - 1] != '/')
+		url << "/";
+	url << ((BStringField *)row->GetField(CLM_INTERNAL_NAME))->String();
 
-	BMessage *clip = (BMessage *)NULL;
 	if (be_clipboard->Lock()) {
 		be_clipboard->Clear();
-		if (clip = be_clipboard->Data()) {
+		BMessage* clip = be_clipboard->Data();
+		if (clip == B_OK) {
```
Comment (owenca) [inline src/FTPWindow.cpp]:
```suggestion
		if (clip != NULL) {
```

File: src/FTPWindow.cpp (hunk 3 above)
Comment (Zotyamester) [reply src/FTPWindow.cpp]: Thanks! Fixed that.

File: src/FTPWindow.cpp (hunk 5 above)
Comment (Zotyamester) [reply src/FTPWindow.cpp]: Oh, silly me! I thought of it as if it returned with a status_t. Any way I fixed it.

**Comment by owenca** [comment]
A few more trivial ones:
https://github.com/HaikuArchives/FtpPositive/blob/b946be43c1d1646f467646af355cae3d5b4a565f/src/FTPWindow.cpp#L414
https://github.com/HaikuArchives/FtpPositive/blob/b946be43c1d1646f467646af355cae3d5b4a565f/src/FTPWindow.cpp#L426-L435
https://github.com/HaikuArchives/FtpPositive/blob/b946be43c1d1646f467646af355cae3d5b4a565f/src/FTPWindow.cpp#L665-L666
https://github.com/HaikuArchives/FtpPositive/blob/b946be43c1d1646f467646af355cae3d5b4a565f/src/FTPWindow.cpp#L722-L725

**Comment by Zotyamester** [comment]
Done

File: src/FTPWindow.cpp [hunk 6]
```diff
@@ -664,8 +657,6 @@ void TFTPWindow::DirlistChanged(BMessage *msg)
 	fRemoteFileView->Clear();
 
 	BString xpwd_result;
-	if (msg->FindString("xpwd", &xpwd_result) != B_OK) {
```
Comment (owenca) [inline src/FTPWindow.cpp]: `xpwd_result` is used below. Otherwise, you could have deleted its declaration as well.

**Comment by humdingerb** [comment]
Thanks guys!

**Pull request merged**